src/compiler/checker.ts

```diff
@@ -17967,7 +17967,10 @@ namespace ts {
 
             for (let i = 0; i < paramCount; i++) {
                 const sourceType = i === restIndex ? getRestTypeAtPosition(source, i) : tryGetTypeAtPosition(source, i);
-                const targetType = i === restIndex ? getRestTypeAtPosition(target, i) : tryGetTypeAtPosition(target, i);
+                let targetType = i === restIndex ? getRestTypeAtPosition(target, i) : tryGetTypeAtPosition(target, i);
+                if (i === restIndex && targetType && sourceType && isTupleType(sourceType) && !sourceType.target.hasRestElement) {

```

Comment:
I have concluded that `sourceType.target.hasRestElement` doesn't have to come with any special handling here but perhaps there is a case that could be handled here - if you come up with any I might try to accommodate it in the logic.